LocalStorage/LocalStorageManager/LocalStorageManagerProtocol.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  LocalStorageManagerProtocol.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 06/04/2024.
+//
+
+import Foundation
+import Combine
+
+protocol LocalStorageManagerProtocol {
+    static var shared: LocalStorageManager {get}
+    
+    func getCities() -> [LocalStorageCity]
+    func addCity(_ weatherInfo: FormattedCityWeatherModel) throws
+    func deleteCity(by name: String) throws
+    func clearDatabase() throws
+    
+    func changeLocalStorageType(_ localDb: LocalStorageProtocol) -> LocalStorageProtocol
+    
+    /// outputs
+    var cities: PassthroughSubject<Result<[LocalStorageCity], LocalStorageError>, Never> {get}
```

Comment:
```suggestion
```
This violated the single responsibility prinsible. A local storage manager should only be responsible for adding, updating, and removing the models from a storage and it shouldn't be aware of this observation which could be added to another layer.